src/main/webapp/query-control.js

```diff
@@ -216,6 +224,75 @@ async function createStateDropdownList() {
   });
 }
 
+// Set up an autocomplete dropdown, attached to the main location input
+// and states dropdown, to suggest places to users as they type
+function setupAutocompleteLocation() {
+  const autocompleteInput = document.getElementById('location-list');
+  const autocompleteResults = document.getElementById('location');
+  const service = new google.maps.places.AutocompleteService();
+  const defaultLocationOptions = autocompleteResults.innerHTML; // all states
+
+  function displaySuggestions(predictions, status) {
+    if (status != google.maps.places.PlacesServiceStatus.OK) {
+      alert(
```

Comment:
When I start with the default location, click the dropdown, and click the default each U.S. state again, I get the zero results error. (Or if I just try typing "each "). Is there a way to not have the alert pop up for zero results and just show the dropdown with only states? Also, the input clears when there are no autocomplete results, though the input would be a business if it doesn't show in the suggestions, I'm wondering if users should be allowed to input anything?

Reply:
Ah, that's a good point. I can just delete this if-clause and leave the next one that returns when there aren't any predictions. 
I'm not sure what you mean by your second question, sorry, could you elaborate?

Reply:
Oh after checking again the input comes back after you click away, as in if I type a non-location in, the zero results alert pops up, then I click ok, then the location input is blank, but then I click somewhere else and the non-location I typed comes back. Was wondering if the input can come back right after clicking OK

Reply:
Ah, gotcha. I think I deleted the alert entirely, so this shouldn't be a problem any more. I'll deploy the latest version now so you can take a look.

Reply:
Looks and works great now, thanks!